README.md

```diff
@@ -95,6 +95,42 @@ You can use a service like [Polyfill.io](https://polyfill.io) to download only t
 <script src="https://cdn.polyfill.io/v2/polyfill.min.js?features=Element.prototype.classList"></script>
 ```
 
+### Shadow DOM
+
+It could be very expensive to apply this polyfill automatically to every shadow
+root that is created in a given document, so the polyfill ignores shadow roots
+by default. If you are using Shadow DOM in a component, it is possible to apply
+this polyfill imperatively to the component's shadow root:
+
+```javascript
+// Check for the polyfill:
+if (self.applyFocusVisiblePolyfill != null) {
+  self.applyFocusVisiblePolyfill(myComponent.shadowRoot);
+}
+```
+
+### Lazy-loading
+
+When this polyfill is lazy-loaded, and you are applying the polyfill to a shadow
+root with JavaScript, it is important to know when the polyfill has become
+available before trying to use it.
+
+In order to act at the right time, you can observe the global
+`focus-visible-polyfill-ready` event:
+
+```javascript
+window.addEventListener('focus-visible-polyfill-ready',
+    () => self.applyFocusVisiblePolyfill(myComponent.shadowRoot),
```

Comment:
Similar to above, what's the expected value of `self` in this context?

Reply:
Will change to `window`.